ui/src/TestAzureAccount.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { SubscriptionClient } from 'azure-arm-resource';
+import { SubscriptionListResult, TenantListResult } from 'azure-arm-resource/lib/subscription/models';
+import { ApplicationTokenCredentials, AzureEnvironment, loginWithServicePrincipalSecret } from 'ms-rest-azure';
+import { Event, EventEmitter } from 'vscode';
+import { AzureAccount, AzureLoginStatus, AzureResourceFilter, AzureSession, AzureSubscription } from './azure-account.api';
+import { ArgumentError } from './errors';
+import { localize } from './localize';
+
+export class TestAzureAccount implements AzureAccount {
+    public status: AzureLoginStatus;
+    public onStatusChanged: Event<AzureLoginStatus>;
+    public waitForLogin: () => Promise<boolean>;
+    public sessions: AzureSession[];
+    public onSessionsChanged: Event<void>;
+    public subscriptions: AzureSubscription[];
+    public onSubscriptionsChanged: Event<void>;
+    public waitForSubscriptions: () => Promise<boolean>;
+    public filters: AzureResourceFilter[];
+    public onFiltersChanged: Event<void>;
+    public waitForFilters: () => Promise<boolean>;
+    private onStatusChangedEmitter: EventEmitter<AzureLoginStatus>;
+    private onFiltersChangedEmitter: EventEmitter<void>;
+
+    public constructor() {
+        this.subscriptions = [];
+        this.status = 'Initializing';
```

Comment:
Not sure it should stay in the Initializing state, I think it should switch to LoggedOut just like the real one.  Probably after a very small delay (just make sure testLogIn() waits for initialization to succeed first so that test code that immediately asks for login doesn't fail - this mimics the fact that the real one won't let users log in until it's done initializing).